config/webpack.config.dev.js

```diff
@@ -11,7 +11,7 @@ var path = require('path');
 var autoprefixer = require('autoprefixer');
 var webpack = require('webpack');
 var HtmlWebpackPlugin = require('html-webpack-plugin');
-var paths = require('./paths');
+var paths = require('./')('paths');
```

Comment:
I agree that the code is confusing to some extent.  In fact, I leant very recently that `require` looks for `index.js` when the required path is a folder.

Two ideas:

1.  replace it with `require('../config')('paths');` even if the file is in the same folder.  At least the word `config` appears in the code to suggest the meaning of this line.

2. create a *symbolic name* **config** so that the above code is rewritten as `require('config')('paths')`.

I don't know how to do 2 honestly.  I will take a look at [this method](https://gist.github.com/branneman/8048520#1-the-symlink) and search for others.

There is another decision worthy a discussion: should `config/index.js` export a function (like the current change) or an object with fields like `paths` and so on.

It is more conventional for a module to export an object or a constructor.  But exporting a function has advantages in this case:

1.  the code is short, and we don't need to introduce a new field every time when a new configuration is introduced; simply create a *.prod|dev.js file.

2. we could add an error message/exception when developers made a typo on importing a configuration, let the typo be exposed earlier.   

3. in test/dev, we are still able to import a prod config (e.g. the current `build.js` always uses `webpack.prod`)